oci/spec_opts_unix.go

```diff
@@ -68,13 +65,15 @@ func WithLinuxNamespace(ns specs.LinuxNamespace) SpecOpts {
 }
 
 // WithImageConfig configures the spec to from the configuration of an Image
-func WithImageConfig(i Image) SpecOpts {
-	return func(ctx context.Context, client *Client, c *containers.Container, s *specs.Spec) error {
-		var (
-			image = i.(*image)
-			store = client.ContentStore()
-		)
-		ic, err := image.i.Config(ctx, store, platforms.Default())
+func WithImageConfig(image Image) SpecOpts {
+	return func(ctx context.Context, client Client, c *containers.Container, s *specs.Spec) error {
+		store := client.ContentStore()
+		// TODO: needs review. Previous this passed in a store, now it's using
+		// a store that is part of the client. This makes the assumption that
+		// the client store is the same as the one wrapped by `image`.
+		// Is this assumption safe? Should the interface be changed to accept
+		// a store instead of wrapping one?
+		ic, err := image.Config(ctx)
```

Comment:
This needs careful review. I believe it's correct now, but I'm not really happy with the assumption it makes.

It seems like it would be safer to accept a store parameter on `image.Config()` however that also seems like it defeats the purpose of the `containerd.Image` interface.

The same change was made to the windows version.